docs: improve main doc (#11)
benchttp/infra: README.md

This repository holds the code for our infrastructure and GitHub repositories.

## Repository structure

Each infrastructure block is defined in a directory with Terraform modules.

Inside you will find:

- the entry point inside `main.tf`
- the expected variables to configure the infrastructure inside `variables.tf`

> - a mock of the actual variables definition inside `env.tfvars.example`, not versioned for obvious reasons

Reviewer: ```suggestion
- an example of the actual variables definition inside `env.tfvars.example`
```

## Usage

### Set up your local environment

Move to the directory of the desired block:

```sh
cd block
```

> Rename copy `env.tfvars.example` to `terraform.tfvars` and replace its values with the correct ones.

Reviewer: Let's keep the naming consistent, so either:
> `env.tfvars.example` ➡️ `env.tfvars`

or

> `terraform.tfvars.example` ➡️ `terraform.tfvars`

```sh
cp env.tfvars.example terraform.tfvars
```

By using your own environment, you can deploy an exact copy of the solution as your own. To work with Benchttp original solution, reach one of the repository owners.

### Deploy the infrastructure

Initialize Terraform if not done already:

```sh
terraform init

> ```

Reviewer: Perhaps we should explicit the fact that this is done at a block scope.

Plan the infrastructure without actually making any real world changes (i.e. a dry run):

```sh
make plan
```

Apply the infrastructure and introduce changes to the currently deployed system:

```sh
make apply

> ```

Reviewer: This is outdated, these commands don't work anymore since the architectural change.
Out of scope, but the `Makefile` was a bit silly honestly, we probably should replace it with regular bash scripts in a dedicated directory.

> Refer to the [Terratform docs](https://www.terraform.io/docs) for further documentation.

Reviewer: ```suggestion
Refer to the [Terraform docs](https://www.terraform.io/docs) for further documentation.
```
